plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -198,6 +198,132 @@ Patch: "something"
 	})
 }
 
+func TestPatchTransformerNotAllowedMultipleStrategicMergePatches(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+
+	th.WriteF(`multiplepatches.yaml`, `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myDeploy
+spec:
+  template:
+    spec:
+      containers:
+      - image: AUDIT_IMAGE
+        name: manager
+        args:
+        - --port=8443
+        - --logtostderr
+        - --emit-admission-events
+        - --exempt-namespace=gatekeeper-system
+        - --operation=webhook
+        - --disable-opa-builtin=http.send
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: yourDeploy
+spec:
+  template:
+    spec:
+      containers:
+      - image: CONTROLLER_IMAGE
+        name: manager
+        args:
+        - --emit-audit-events
+        - --operation=audit
+        - --operation=status
+        - --logtostderr`)
+
+	th.RunTransformerAndCheckError(`
+apiVersion: builtin
+kind: PatchTransformer
+metadata:
+  name: notImportantHere
+Path: multiplepatches.yaml
+target:
+  name: .*Deploy
+  kind: Deployment
+`, someDeploymentResources, func(t *testing.T, err error) {
+		t.Helper()
+		if err == nil {
+			t.Fatalf("expected error")
+		}
+		if !strings.Contains(err.Error(),
+			"Multiple Strategic-Merge Patch file in one 'patches' entry is not allowed to set 'patches.target' field.") {
+			t.Fatalf("unexpected err: %v", err)
+		}
+	})
+}
+
+func TestPatchTransformerNotAllowedMultipleJsonPatches(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+	oneDeployment := `
+apiVersion: apps/v1
+metadata:
+  name: oneDeploy
+kind: Deployment
+spec:
+  replica: 1
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.7.9
+      - name: sidecar
+        image: busybox:1.36.1
+`
+	config := `
+apiVersion: builtin
+kind: PatchTransformer
+metadata:
+  name: notImportantHere
+patch: |-
+  - op: replace
+    path: /spec/template/spec/containers/0/image
+    value: nginx:latest
+  ---
+  - op: replace
+    path: /spec/template/spec/containers/1/image
+    value: busybox:latest
+target:
+  name: .*Deploy
+  kind: Deployment
+`
+
+	// TODO(5049): Multiple JSON patch Yaml documents are not allowed and need to return error, but we don't implement it yet.
```

Comment:
nit: Can we wrap this comment at 80-100 characters to better line up with the code?

Reply:
Yes, I will be short this `TODO` comment to fit into one line.